scripts/startup.sh

```diff
@@ -226,29 +228,252 @@ case ${options[$?]} in
+    WARNING: ARCH LINUX WILL BE INSTALLED IN THE SELECTED PARTITION.
+    This will delete all the data present in it, so make sure that
+    you select the right partition that donot contain any useful files.
+------------------------------------------------------------------------
+
+
+"
+
+  fdisk -l
+
+  echo -e "\n"
+  echo "Select the Partition to Install on (This will be the root partition mounted on \"\\\"): "
+  options=($(lsblk -n --output TYPE,KNAME,SIZE,MOUNTPOINT | awk '$1=="part"{print "/dev/"$2"|"$3"__________"$4}'))
+  
+  select_option $? 1 "${options[@]}"
+  part=${options[$?]%|*}
+  set_option INSTALL_IN "PART"
+  set_option DISK ""
+  set_option PART ${part%|*}
+
+  echo -e "\n${part%|*} selected as root partition. Is that OK?"
+  options=("Yes" "No")
+  select_option $? 1 "${options[@]}"
+
+  case ${options[$?]} in
+  y|Y|yes|Yes|YES)
+    drivessd ;;
+  n|N|no|NO|No)
+    diskpart
+    return
+  ;;
+  *)
+    diskpart
+    return
+  ;;
+  esac
+
+  if [[ -d "/sys/firmware/efi" ]]; then # Checking for UEFI System
+    clear
+    echo -en "
+    ------------------------------------------------------------------------
+        NOTE: If you had installed Windows or any other OS in the current
+        disk drive, an EFI System partition may already exist in it.
+        Identify it from the table below and select it. But, if it doesnot
+        exist, then create a partition of size 100 MB and set its type to
+        EFI System and choose \"yes\" when asked if you want to format it.
+    ------------------------------------------------------------------------
+
+    "
+
+    fdisk -l
+
+    echo -e "\n"
+    echo "Select the EFI Partition (This partition should be in FAT32 format and about 100-500 MB in size. It will be mounted on \"\\Boot\\EFI\"): "
+    options=("Create" $(lsblk -n --output TYPE,KNAME,SIZE,MOUNTPOINT | awk '$1=="part"{print "/dev/"$2"|"$3"__________"$4}'))
+  
+    select_option $? 1 "${options[@]}"
+    case $? in
+    0)
+      manualpart
+      diskpart
+      return
+    ;;
+    *)
+      BOOTpart=${options[$?]%|*}
+      set_option BOOTPART ${BOOTpart%|*}
+    ;;
+    esac
+
+    echo -e "\n${BOOTpart%|*} selected as EFI partition. Is that OK?"
+    options=("Yes" "No")
+    select_option $? 1 "${options[@]}"
+    case ${options[$?]} in
+    y|Y|yes|Yes|YES)
+      return ;;
+    n|N|no|NO|No)
+      diskpart
+      return
+    ;;
+    *)
+      diskpart
+      return
+    ;;
+    esac
+
+    echo -e "\nDo you want to format the selected EFI partition?"
+    options=("Yes" "No")
+    select_option $? 1 "${options[@]}"
+    case ${options[$?]} in
+    y|Y|yes|Yes|YES) set_option FORMATEFI "yes" ;;
+    n|N|no|NO|No) set_option FORMATEFI "no" ;;
+    *) set_option FORMATEFI "no" ;;
+    esac
+
+  elif [[ $(sudo sudo fdisk -l | grep -i '^Disklabel type') = "Disklabel type: gpt" ]]; then # Checking for GPT Disk Label on a Legacy BIOS (non UEFI) System
```

Comment:
there are two typos
1. sudo sudo
2.  '^Disklabel type' contains  )

I wasnt able to produce result of this command or may be  using
(sudo fdisk -l | grep -i  "Disklabel type: gpt") -ne "0"

Reply:
1. Actually theres no need for sudo as this scipt itself will run with sudo and I somehow typed it twice.

2. In my PC, the command works both with and without the "^"

I just realised that this if statement will be true even if only one of many disks in the device is GPT. I will have to change it to check the drive selected by the user only.

Reply:
@mfgbhatti, just asking, please let me know if you know a reliable way to get the disk name a partition is present in. It can not be just a string manipulation because SSDs and NVMes can have disk name as '\dev\nvme0' and partition name as '\dev\nvme0p1', as a result just removing the number from the end of the partition name is not enough.

Reply:
> I just realised that this if statement will be true even if only one of many disks in the device is GPT. I will have to change it to check the drive selected by the user only.

https://github.com/ChrisTitusTech/ArchTitus/commit/4eeaa78c075093b7a8c78c79c71fb4431d10d79a solves this. See https://github.com/ChrisTitusTech/ArchTitus/pull/293#discussion_r836027063.